pkg/resources/helpers.go

```diff
@@ -330,3 +330,183 @@ func ListDiff[T comparable](beforeList []T, afterList []T) (added []T, removed [
 
 	return added, removed
 }
+
+func StorageLocationsEqual(s1 sdk.ExternalVolumeStorageLocation, s2 sdk.ExternalVolumeStorageLocation) (bool, error) {
```

Comment:
It would be great to check, if we can use `DeepEqual` here. If not, I think we should at least compare on the provider level (meaning, add `Equal` function) to e.g. `sdk.S3StorageLocationParams` and call it here.